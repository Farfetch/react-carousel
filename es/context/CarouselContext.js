import React from 'react';
export default /*#__PURE__*/React.createContext({
  goNext: function goNext() {
    return null;
  },
  goPrev: function goPrev() {
    return null;
  },
  goTo: function goTo() {
    return null;
  },
  setActiveItem: function setActiveItem() {
    return null;
  },
  setItemsLength: function setItemsLength() {
    return null;
  },
  setIsMovementBlocked: function setIsMovementBlocked() {
    return null;
  },
  activeItem: 0,
  isMovementBlocked: false,
  isRTL: false,
  itemsToShow: 0,
  itemsLength: 0,
  itemsToScroll: 0,
  direction: null
});