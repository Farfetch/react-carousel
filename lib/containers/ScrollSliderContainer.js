"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _context = require("../context");

var _components = require("../components");

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var ScrollSliderContainer = function ScrollSliderContainer(props) {
  return /*#__PURE__*/_react["default"].createElement(_context.CarouselContext.Consumer, null, function (_ref) {
    var activeItem = _ref.activeItem,
        isRTL = _ref.isRTL,
        itemsToShow = _ref.itemsToShow,
        setItemsLength = _ref.setItemsLength,
        goTo = _ref.goTo;
    return /*#__PURE__*/_react["default"].createElement(_components.ScrollSlider, _extends({}, props, {
      isRTL: isRTL,
      activeItem: activeItem,
      itemsToShow: itemsToShow,
      setItemsLength: setItemsLength,
      goTo: goTo
    }));
  });
};

var _default = ScrollSliderContainer;
exports["default"] = _default;
module.exports = exports.default;