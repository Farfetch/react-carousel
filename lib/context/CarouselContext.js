"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = /*#__PURE__*/_react["default"].createContext({
  goNext: function goNext() {
    return null;
  },
  goPrev: function goPrev() {
    return null;
  },
  goTo: function goTo() {
    return null;
  },
  setActiveItem: function setActiveItem() {
    return null;
  },
  setItemsLength: function setItemsLength() {
    return null;
  },
  setIsMovementBlocked: function setIsMovementBlocked() {
    return null;
  },
  activeItem: 0,
  isMovementBlocked: false,
  isRTL: false,
  itemsToShow: 0,
  itemsLength: 0,
  itemsToScroll: 0,
  direction: null
});

exports["default"] = _default;
module.exports = exports.default;