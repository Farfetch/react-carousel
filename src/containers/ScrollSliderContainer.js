import { CarouselContext } from '#context';
import { ScrollSlider } from '#components';
import React from 'react';

const ScrollSliderContainer = (props) => (
    <CarouselContext.Consumer>
        {({ activeItem, isRTL, itemsToShow, setItemsLength, goTo }) => (
            <ScrollSlider
                {...props}
                isRTL={isRTL}
                activeItem={activeItem}
                itemsToShow={itemsToShow}
                setItemsLength={setItemsLength}
                goTo={goTo}
            />
        )}
    </CarouselContext.Consumer>
);

export default ScrollSliderContainer;
