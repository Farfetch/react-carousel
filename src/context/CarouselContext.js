import React from 'react';

export default React.createContext({
    goNext: () => null,
    goPrev: () => null,
    goTo: () => null,
    setActiveItem: () => null,
    setItemsLength: () => null,
    setIsMovementBlocked: () => null,
    activeItem: 0,
    isMovementBlocked: false,
    isRTL: false,
    itemsToShow: 0,
    itemsLength: 0,
    itemsToScroll: 0,
    direction: null,
});
