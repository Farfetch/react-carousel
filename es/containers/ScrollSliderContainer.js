function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { CarouselContext } from "../context";
import { ScrollSlider } from "../components";
import React from 'react';

var ScrollSliderContainer = function ScrollSliderContainer(props) {
  return /*#__PURE__*/React.createElement(CarouselContext.Consumer, null, function (_ref) {
    var activeItem = _ref.activeItem,
        isRTL = _ref.isRTL,
        itemsToShow = _ref.itemsToShow,
        setItemsLength = _ref.setItemsLength,
        goTo = _ref.goTo;
    return /*#__PURE__*/React.createElement(ScrollSlider, _extends({}, props, {
      isRTL: isRTL,
      activeItem: activeItem,
      itemsToShow: itemsToShow,
      setItemsLength: setItemsLength,
      goTo: goTo
    }));
  });
};

export default ScrollSliderContainer;