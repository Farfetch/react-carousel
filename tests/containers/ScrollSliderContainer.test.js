import { createSerializer } from 'enzyme-to-json';
import { mount } from 'enzyme';
import CarouselContext from '../../src/context/CarouselContext';
import React from 'react';
import ScrollSliderContainer from '../../src/containers/ScrollSliderContainer';

const mockContextCarousel = {
    activeItem: 1,
    isRTL: false,
    itemsToShow: 2,
    setItemsLength: jest.fn(),
    goTo: jest.fn(),
    containerRef: {
        current: null,
    },
};

const mockChildren = <div />;

beforeAll(() => {
    expect.addSnapshotSerializer(createSerializer());
});

describe('<ScrollSliderContainer/>', () => {
    it('should render correctly', () => {
        const tree = mount(
            <CarouselContext.Provider value={mockContextCarousel}>
                <ScrollSliderContainer>{mockChildren}</ScrollSliderContainer>
            </CarouselContext.Provider>
        );

        expect(tree).toMatchSnapshot();
    });

    it('should render correctly with custom Props', () => {
        const mockTheme = {
            container: 'container',
        };
        const tree = mount(
            <CarouselContext.Provider value={mockContextCarousel}>
                <ScrollSliderContainer data-testId="test" theme={mockTheme}>
                    {mockChildren}
                </ScrollSliderContainer>
            </CarouselContext.Provider>
        );

        expect(tree.find('ScrollSlider').prop('data-testId')).toBe('test');
        expect(tree.find('ScrollSlider').prop('theme')).toEqual(mockTheme);
    });

    it('should render correctly with correct Context', () => {
        const tree = mount(
            <CarouselContext.Provider value={mockContextCarousel}>
                <ScrollSliderContainer>{mockChildren}</ScrollSliderContainer>
            </CarouselContext.Provider>
        );

        expect(tree.find('ScrollSlider').prop('activeItem')).toBe(
            mockContextCarousel.activeItem
        );
        expect(tree.find('ScrollSlider').prop('isRTL')).toBe(
            mockContextCarousel.isRTL
        );
        expect(tree.find('ScrollSlider').prop('itemsToShow')).toBe(
            mockContextCarousel.itemsToShow
        );
        expect(tree.find('ScrollSlider').prop('setItemsLength')).toBe(
            mockContextCarousel.setItemsLength
        );
        expect(tree.find('ScrollSlider').prop('goTo')).toBe(
            mockContextCarousel.goTo
        );
    });
});
