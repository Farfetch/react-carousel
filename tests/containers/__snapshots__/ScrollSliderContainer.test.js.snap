// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollSliderContainer/> should render correctly 1`] = `
<ScrollSlider
  activeItem={1}
  goTo={[MockFunction]}
  isRTL={false}
  itemsToShow={2}
  setItemsLength={[MockFunction]}
>
  <div />
</ScrollSlider>
`;
